src/app/clusters/occupancy-sensor-server/occupancy-sensor-server.cpp

```diff
@@ -0,0 +1,202 @@
+
+//******************************************************************************
+// Plugin init function
+//******************************************************************************
+void emberAfOccupancySensingClusterServerInitCallback(chip::EndpointId endpoint)
+{
+    HalOccupancySensorType deviceType;
+
+    checkForReportingConfig();
+    deviceType = halOccupancyGetSensorType(endpoint);
+
+    emberAfWriteAttribute(endpoint, ZCL_OCCUPANCY_SENSING_CLUSTER_ID, ZCL_OCCUPANCY_SENSOR_TYPE_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+                          (uint8_t *) &deviceType, ZCL_ENUM8_ATTRIBUTE_TYPE);
+
+    uint8_t deviceTypeBitmap = 0;
+    switch (deviceType)
+    {
+    case HAL_OCCUPANCY_SENSOR_TYPE_PIR:
+        deviceTypeBitmap = EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_PIR;
+        break;
+
+    case HAL_OCCUPANCY_SENSOR_TYPE_ULTRASONIC:
+        deviceTypeBitmap = EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_ULTRASONIC;
+        break;
+
+    case HAL_OCCUPANCY_SENSOR_TYPE_PIR_AND_ULTRASONIC:
+        deviceTypeBitmap = (EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_PIR | EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_ULTRASONIC);
+        break;
+
+    case HAL_OCCUPANCY_SENSOR_TYPE_PHYSICAL:
+        deviceTypeBitmap = EMBER_AF_OCCUPANCY_SENSOR_TYPE_BITMAP_PHYSICAL_CONTACT;
+        break;
+
+    default:
+        break;
+    }
+    emberAfWriteAttribute(endpoint, ZCL_OCCUPANCY_SENSING_CLUSTER_ID, ZCL_OCCUPANCY_SENSOR_TYPE_BITMAP_ATTRIBUTE_ID,
+                          CLUSTER_MASK_SERVER, (uint8_t *) &deviceTypeBitmap, ZCL_BITMAP8_ATTRIBUTE_TYPE);
+
+    emberAfPluginOccupancyClusterServerPostInitCallback(endpoint);
+}
+
+//******************************************************************************
+// Notification callback from the HAL plugin
+//******************************************************************************
+void halOccupancyStateChangedCallback(EndpointId endpoint, HalOccupancyState occupancyState)
+{
+    if (occupancyState == HAL_OCCUPANCY_STATE_OCCUPIED)
+    {
+        emberAfOccupancySensingClusterPrintln("Occupancy detected");
+    }
+    else
+    {
+        emberAfOccupancySensingClusterPrintln("Occupancy no longer detected");
+    }
+
+    emberAfWriteAttribute(endpoint, ZCL_OCCUPANCY_SENSING_CLUSTER_ID, ZCL_OCCUPANCY_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+                          (uint8_t *) &occupancyState, ZCL_BITMAP8_ATTRIBUTE_TYPE);
+    // emberAfPluginOccupancySensorServerOccupancyStateChangedCallback(
+    //   endpoint, occupancyState);
+}
+
+void emberAfPluginOccupancySensorServerStackStatusCallback(EndpointId endpoint, EmberStatus status)
+{
+    // On network connect, verify that a reporting entry is set up for the
+    // occupancy sensor
+    if (status == EMBER_NETWORK_UP)
+    {
+        checkForReportingConfig();
+    }
+}
+
+//------------------------------------------------------------------------------
+// Plugin private functions
+
+static void checkForReportingConfig(void)
```

Comment:
This whole function should presumably go away.

Reply:
done.